### PR TITLE
fix(`order by`): for localized sorting, prepend table alias

### PR DESCRIPTION
**Pull request by patricebender**
Usually, column references in _simple_ names in order by clauses are left as is. That means **no table alias** is prepended. In expressions however, **only source elements** can be reffered to.

That is why in case of localized sorting, we must prepend the table alias nevertheless, as in the DB specific drivers, a `COLLATE` expression is wrapped around the string reference. Hence we must replace the column reference with the underlying source reference.

as reported in #543, postgres is not able to resolve a reference to a column unambigously if the reference is part of a `COLLATE` expression.

### CODE REVIEW

**Comment by patricebender** [comment]
@stewsk @BobdenOs please have a look

File: test/scenarios/bookshop/orderBy.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    // the original query is sorted by the **column** "createdBy"
+    // the resulting query has two query sources in the end --> Authors and Books
+    // even though both Tables have the element "createdBy", the DB should be able to resolve the
+    // order by reference to the column unambiguously
+    const query = CQL(`SELECT from sap.capire.bookshop.Books {
+      createdBy,
+      author.name as author
+    } order by createdBy`)
+    const res = await cds.run(query)
+    expect(res.length).to.be.eq(5)
```
Comment (BobdenOs) [inline test/scenarios/bookshop/orderBy.test.js:17]: I would recommend putting a limit on the query and expecting `1`. As it is order relevant it might make sense to check for the returned value. As currently it just checks that it does not throw.

File: test/scenarios/bookshop/orderBy.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  })
+  test('collations with val', async () => {
+    if(cds.env.requires.db.impl === '@cap-js/hana')
+      return // FIXME: the `val` is put into window function, which ends in an error on HANA
```
Comment (BobdenOs) [inline test/scenarios/bookshop/orderBy.test.js:21]: Will have to look what can be done about that. Luckily sorting by static values is not the most common. Might even make sense to simply filter them out for `HANA`.

File: .gitignore [hunk 3]
```diff
@@ -122,6 +122,9 @@ dist
 # Stores VSCode versions used for testing VSCode extensions
 .vscode-test
 
+# VSCode configs
+/.vscode
```
Comment (BobdenOs) [inline .gitignore:126]: This seems a bit unrelated to this change. Are there some vscode configs we want to include in this repository ?

Comment (patricebender) [reply .gitignore]: it might make sense to enforce prettier as formatter

File: .gitignore [hunk 4]
```diff
@@ -122,6 +122,7 @@ dist
 # Stores VSCode versions used for testing VSCode extensions
 .vscode-test
 
+
```
Comment (patricebender) [inline .gitignore:125]:
```suggestion
```

**Review by BobdenOs — APPROVED** [review]
